website/static/js/comment.js

```diff
@@ -132,29 +146,113 @@ BaseComment.prototype.setupToolTips = function(elm) {
     });
 };
 
-BaseComment.prototype.fetch = function() {
+BaseComment.prototype.fetch = function(isCommentList, thread) {
     var self = this;
     var deferred = $.Deferred();
     if (self._loaded) {
         deferred.resolve(self.comments());
     }
+    if (thread !== undefined) {
+        return self.getThread(thread);
+    }
     $.getJSON(
         nodeApiUrl + 'comments/',
-        {target: self.id()},
+        {
+            page: self.page(),
+            target: self.id(),
+            rootId: self.rootId(),
+            isCommentList: (isCommentList || null)
+        },
         function(response) {
             self.comments(
-                ko.utils.arrayMap(response.comments.reverse(), function(comment) {
+                ko.utils.arrayMap(response.comments.reverse(), function (comment) {
                     return new CommentModel(comment, self, self.$root);
                 })
             );
+            if (isCommentList) {
+                self.discussionByFrequency(response.discussionByFrequency);
+                self.discussionByRecency(response.discussionByRecency);
+            }
             self.unreadComments(response.nUnread);
             deferred.resolve(self.comments());
+            self.checkFileExists();
             self._loaded = true;
         }
     );
     return deferred;
 };
 
+BaseComment.prototype.getThread = function(thread_id) {
+    var self = this;
+    var deferred = $.Deferred();
+    if (self._loaded) {
+        deferred.resolve(self.comments());
+    }
+    $.getJSON(
+        nodeApiUrl + 'comment/' + thread_id + '/',
+        {},
+        function(response) {
+            self.comments([]);
+            self.comments.push(new CommentModel(response.comment, self, self.$root));
+            deferred.resolve(self.comments());
+            self.checkFileExists();
+            self._loaded = true;
+        }
+    );
+    return deferred;
+}
+
+BaseComment.prototype.checkFileExists = function() {
```

Comment:
This name is misleading, as this method does not just do a "check" that returns a boolean; it also sets fields of comments. 

Can this be broken down into multiple methods/functions?